app/src/main/java/es/situm/gettingstarted/drawbuilding/DrawBuildingActivity.java

```diff
@@ -31,10 +38,15 @@
         extends SampleActivity
         implements OnMapReadyCallback {
 
-
     private GoogleMap map;
     private Building building;
+    private FloorAdapter floorAdapter;
+    private ArrayList<Floor> reverseList;
```

Comment:
Intenta usar interfaces de alto nivel nas declaracións.
Neste caso sería `private List<Floor> floors;`
O nome "reverseList" non me gusta demasiado, explícome: neste exemplo amosamos como dibuxar no mapa os diferentes planos dun edificio, cada un referido a unha planta. Para un desarrollador que acaba de chegar ao exemplo, "reverseList" non sei si lle di algo. En cambio, utilizaría outro nome como "buildingFloors" ou "floors" ou o que vexas que pode ser máis intuitivo.

Reply:
Podes declarar un array list internamente nunha clase, Outra cousa sería que fose a o parámetro dun método. Igual precisas que ese obxecto dentro da tua clase teña os métodos de ArrayList.